undefined reference to `TLS_server_method'

Commit 64fc55d85b4fe5b3b3a26b1b99f30b9201c455d8 broke the build:

```
# ...
/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: src/websocket.o: in function `ws_start':
websocket.c:(.text+0x325e): undefined reference to `TLS_server_method'
collect2: error: ld returned 1 exit status
```

I build GoAccess on a shared webspace, so I don't have *root*. Anyway, the webspace runs CentOS 7 which will be supported until 2024, so it's fine:

```
$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
# ...

$ uname -sr
Linux 3.10.0-1160.31.1.el7.x86_64

$ openssl version
OpenSSL 1.0.2k-fips  26 Jan 2017
```

My build file is as follows:

```shell
#!/bin/sh
cd $HOME/git/goaccess
make clean
git fetch ; git pull
./configure --enable-utf8 --prefix=$HOME --with-openssl --enable-geoip=legacy
make && make install
```

Maybe there should be a check for *really LTS* versions?

--- a/src/websocket.c
+++ b/src/websocket.c
@@ -689,7 +689,11 @@ initialize_ssl_ctx (WSServer * server) {
   OpenSSL_add_ssl_algorithms ();
 
   /* ssl context */
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+  if (!(ctx = SSL_CTX_new (SSLv23_server_method ())))
+#else
   if (!(ctx = SSL_CTX_new (TLS_server_method ())))
+#endif
     goto out;
   /* set certificate */
   if (!SSL_CTX_use_certificate_file (ctx, wsconfig.sslcert, SSL_FILETYPE_PEM))

